app/assets/javascripts/leaflet.map.js

```diff
@@ -9,6 +14,43 @@ L.extend(L.LatLngBounds.prototype, {
   }
 });
 
+if (OSM.MAPTILER_KEY) {
+  maplibregl.setRTLTextPlugin(
+    "https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-rtl-text/v0.2.3/mapbox-gl-rtl-text.js",
+    true
+  );
+
+  L.OpenMapTiles = L.MaplibreGL.extend({
+    options: {
+      maxZoom: 23,
+      style: "https://api.maptiler.com/maps/openstreetmap/style.json?key=" + OSM.MAPTILER_KEY
+    },
+    onAdd: function (map) {
+      L.MaplibreGL.prototype.onAdd.call(this, map);
+      var m = this.getMaplibreMap();
+      var supportedLanguages = maplibregl.Map.prototype.supportedLanguages;
+      var language;
+      if (supportedLanguages.includes(I18n.locale)) {
+        language = I18n.locale;
+      } else {
+        var mainLocale = I18n.locale.slice(0, 2);
+        var localeIndex = supportedLanguages.findIndex(function (locale) {
+          return locale === mainLocale;
+        });
```

Comment:
This website and maplibre-gl-omt-language both support some locales that have three-character-long language codes, such as [Western Punjabi](https://github.com/openstreetmap/openstreetmap-website/blob/e206dd527ef7c6d1829f131533d722e536e4bfb5/config/locales/pnb.yml) (pnb). I don’t think this code will match such locales. If I’m not mistaken, the intention is to match on only the language code and ignore any script or country code. If so, this would be more robust:

```suggestion
        var mainLocale = I18n.locale.split('-')[0];
        var localeIndex = supportedLanguages.findIndex(function (locale) {
          return locale === mainLocale;
        });
```

Note that this is still quite a rudimentary locale matching heuristic. It won’t match `zh-Hans` or `zh-Hant`, which the plugin also supports.

Reply:
I have reworked language matching.